### DISCUSSION TITLE
Get annotated tags

### DISCUSSION THREAD

**Discussion by AlesyaSeliazniova30032012** [gitpython-developers/GitPython#1937] [Q&A] [answered]
I need to use GItPython API to get annotated tags. If I do next:

`
repo = git.Repo(repo_path)
tags = repo.tags
annotated_tags = [tag for tag in tags]
`
I get list of lightweight tags. How do I get a list of annotated tags? 
I need for annotated tags info: tag, tagger, tagged_date
Thanks for any help

**Comment by Byron** [comment, marked as answer]
Maybe this helps: https://chatgpt.com/share/b1010a46-8079-414e-89ab-da43b2af90b9 .

Here is a copy of the code it produced:

```python
import git

# Path to your repository
repo_path = '/path/to/your/repo'

# Initialize the Repo object
repo = git.Repo(repo_path)

# Get all tags
all_tags = repo.tags

# Filter annotated tags
annotated_tags = [tag for tag in all_tags if tag.tag is not None]

# Print the annotated tags
for tag in annotated_tags:
    print(f"Annotated Tag: {tag.name}")
    print(f"Tag Message: {tag.tag.message}")
    print(f"Tag Object: {tag.tag.object}")

```
